libcalico-go/lib/clientv3/ippool_e2e_test.go

```diff
@@ -318,10 +335,34 @@ var _ = testutils.E2eDatastoreDescribe("IPPool tests", testutils.DatastoreAll, f
 			_, outError = c.IPPools().Get(ctx, name2, options.GetOptions{})
 			Expect(outError).To(HaveOccurred())
 			Expect(outError.Error()).To(ContainSubstring("resource does not exist: IPPool(" + name2 + ") with error:"))
+
+			By("Adding an IPPool with empty string for IPIPMode and VXLANMode and expecting it to be defaulted to 'Never'")
+			res, outError = c.IPPools().Create(ctx, &apiv3.IPPool{
```

Comment:
I don't think this is testing the code branch that you added, because we do defaulting on `Create` here: https://github.com/projectcalico/calico/blob/e20bfee71d71f288386e097bd8d2041da329753d/libcalico-go/lib/clientv3/ippool.go#L431-L435

You need to bypass that logic somehow (maybe using the backend client directly) to ensure the data that gets stored is missing the VXLAN setting, and then confirm that Get() defaults the field.

Reply:
Oh right, let me look into that. Thanks for catching this!

Reply:
I added a new `DebugUnsafeCreateNoDefaults()` method that skips setting defaults and now the test seems to be testing properly. Thanks!